pallets/messages/src/lib.rs

```diff
@@ -186,13 +186,14 @@ pub mod pallet {
 			on_behalf_of: Option<MessageSourceId>,
 			schema_id: SchemaId,
 			cid: Vec<u8>,
-			_payload_length: u32,
+			payload_length: u32,
 		) -> DispatchResultWithPostInfo {
 			let provider_key = ensure_signed(origin)?;
-			let bounded_payload: BoundedVec<u8, T::MaxMessagePayloadSizeBytes> = cid
-				.clone()
-				.try_into()
-				.map_err(|_| Error::<T>::ExceedsMaxMessagePayloadSizeBytes)?;
+			let bounded_payload: BoundedVec<u8, T::MaxMessagePayloadSizeBytes> =
+				(cid.clone(), payload_length)
```

Comment:
should we assign it to that certain Type first so when you are looking how that type is being used you can see that encode and decode. Other concern is that if this is not first converted to that custom type then this might change and the decode part might not and it would cause issue in decoding.

Reply:
Good idea